include/rocksdb/filter_policy.h

```diff
@@ -216,23 +216,33 @@ class FilterPolicy {
 extern const FilterPolicy* NewBloomFilterPolicy(
     double bits_per_key, bool use_block_based_builder = false);
 
-// An EXPERIMENTAL new Bloom alternative that saves about 30% space
-// compared to Bloom filters, with about 3-4x construction time and
-// similar query times. For example, if you pass in 10 for
+// An new Bloom alternative that saves about 30% space compared to
+// Bloom filters, with about 3-4x construction time and similar
+// query times. For example, if you pass in 10 for
 // bloom_equivalent_bits_per_key, you'll get the same 0.95% FP rate
 // as Bloom filter but only using about 7 bits per key. (This
 // way of configuring the new filter is considered experimental
-// and/or transitional, so is expected to go away.)
+// and/or transitional, so is expected to be replaced with a new API.
+// The constructed filters will be given long-term support.)
 //
-// Ribbon filters are ignored by previous versions of RocksDB, as if
-// no filter was used.
+// The space savings of Ribbon filters makes sense for lower (higher
+// numbered; larger; longer-lived) levels of LSM, whereas the speed of
+// Bloom filters make sense for highest levels of LSM. Setting
+// ribbon_starting_level allows for this design. For example,
+// ribbon_starting_level=1 means that Bloom filters will be used in
+// level 0, including flushes, and Ribbon filters elsewhere.
+// ribbon_starting_level=0 means (almost) always use Ribbon.
 //
-// Note: this policy can generate Bloom filters in some cases.
-// For very small filters (well under 1KB), Bloom fallback is by
-// design, as the current Ribbon schema is not optimized to save vs.
-// Bloom for such small filters. Other cases of Bloom fallback should
-// be exceptional and log an appropriate warning.
+// Ribbon filters are compatible with RocksDB >= 6.15.0. Earlier
+// versions reading the data will behave as if no filter was used
+// (degraded performance until compaction rebuilds filters).
+//
+// Note: even with ribbon_starting_level=0, this policy can generate
+// Bloom filters in some cases. For very small filters (well under 1KB),
+// Bloom fallback is by design, as the current Ribbon schema is not
+// optimized to save vs. Bloom for such small filters. Other cases of
+// Bloom fallback should be exceptional and log an appropriate warning.
 extern const FilterPolicy* NewExperimentalRibbonFilterPolicy(
-    double bloom_equivalent_bits_per_key);
+    double bloom_equivalent_bits_per_key, int ribbon_starting_level = 1);
```

Comment:
I think the level specific information should be outside of a specific filter configuration if it's possible.
And I'm not sure if it's exposing too much information about levels.
Could it be like compression that has CompressionType and Bottommost CompressionType:
https://github.com/facebook/rocksdb/blob/524b10bd6ee5fbea8ec9e90dd668c07d0c653371/include/rocksdb/options.h#L200-L206

Reply:
The intention with this transitional API is to provide some manual controls to play with to help inform what the next generation API should look like

Reply:
And it should be easy to maintain support for this transitional API for a while.

Reply:
If it's going to be deprecated, then we should keep the `Experimental` name.
But what would be the future API look like and should we considering the migrating path? For example, if the future API is like compression, which only has bottommost config, then it would be difficult if we already provide level. (For filter, by the way, I think an upmost config may make more sense.)

Reply:
I agree topmost is likely to deserve special treatment but it's easier to try some different things with customers if the API is there.  The new level threshold functionality is already implementable with public APIs, btw. This change just makes it easier. So I'm not too concerned about migration. So maybe keep the Experimental name of you're ok with it